docs/docsite/rst/dev_guide/overview_architecture.rst

```diff
@@ -67,8 +75,70 @@ Here's what a simple playbook looks like::
     - common
     - content
 
-
-Extending Ansible with plug-ins and the API
+Extending Ansible with plugins and the API
 ===========================================
 
-Should you want to write your own, Ansible modules can be written in any language that can return JSON (Ruby, Python, bash, etc). Inventory can also plug in to any datasource by writing a program that speaks to that datasource and returns JSON. There's also various Python APIs for extending Ansible's connection types (SSH is not the only transport possible), callbacks (how Ansible logs, etc), and even for adding new server side behaviors.
+You can write your own specialized Ansible modules, module utilities, or plugins in any language that can return JSON (Ruby, Python, bash, etc). You can write an :ref:`inventory plugin <developing_inventory>` to connect to any datasource that returns JSON. There are also various Python APIs for extending Ansible's connection types (SSH is not the only transport possible), for adding callbacks (how Ansible logs, etc), and even for adding new server side behaviors.
```

Comment:
Only modules can be written in any language.  The module utility functionality can be written in Python or in powershell.  Other plugins must be written in Python.

The JSON restriction is only on modules.

Reply:
I think I'll get rid of this paragraph entirely and put all this info in the specific sections on modules, module_utils, and plugins.

Reply:
The move to the individual sections looks good to me.